### PR TITLE
20220129 First working GUI

### PR DESCRIPTION
**Pull request by Michaeljfang**
Adding a GUI to PyGAAP that uses the back-end API to process files*;
Support for dynamic scaling and 2x DPI;
Bottom status bar.

*only "prettyFormatResults()" is directly used. The other processing functions in API.py were mirrored and modified.

Current issues:
This has not been tested on macOS;
Only plain-text documents (drop-down menu above "Add" in Canonicizers tab does not change settings);
No Batch Documents or AAAC Problems
Incomplete documentation for how to add a new module.

### CODE REVIEW

**Comment by DavidBerdik** [comment]
The GUI has a bit of a usability issue: when selecting an option from the listboxes, the text of the highlighted element is not readable. Perhaps a darker variant of blue should be used for the UI?

image

File: backend/API.py [hunk 1]
```diff
@@ -65,13 +74,18 @@ def runAnalysis(self, analysisMethodString, distanceFunctionString):
...
 		'''Returns a string of the results in a pretty, formatted structure.'''
 		# Build a string the contains general information about the experiment.
 		formattedResults = str(unknownDoc.title) + ' ' + str(unknownDoc.filepath) + "\nCanonicizers:\n"
 		for canonicizer in canonicizers:
 			formattedResults += '\t' + canonicizer + '\n'
-		formattedResults += "Event Driver:\n\t" + eventDriver + "\nAnalysis Method:\n\t" + analysisMethod + " with " + distanceFunc + '\n'
+		if type(eventDrivers)==list:
+			for eventDriver in eventDrivers:
+				formattedResults += '\t' + eventDriver + '\n'
+		else:
```
[the remark this answers is not in the document]
Comment (Michaeljfang) [reply backend/API.py:86]: This is to make it compatible with the CLI, which expects it to be a string (1 event driver)

[the remark this answers is not in the document]
Comment (DavidBerdik) [reply backend/API.py:86]: Understood. Thank you.

**Comment by DavidBerdik** [comment]
To make it easier for me to work with your code changes, I am merging this PR in to a separate "Michael 3/12/2022 code freeze" branch. This way, I can make any desired changes to your code without interfering with your ongoing development as I do so. Once I complete that process, I will merge the code from this branch in to `master`.

**Pull request merged**